### PR TITLE
OpcodeDispatcher: Optimize cmov 

### PR DESCRIPTION
**Pull request by Sonicadvance1**
cmov was quite terrible in its implementation. Some things of note:
- NZCV cache would cause store for no reason
- {16,32}-bit would zero extend sources for no reason
- 16-bit would zero extend result for no reason

A bunch of flag testing is still doing a ubfx plus compare against zero
when it could end up being a tst instead, but this is a step in the
right direction and switches over to explicit sized selects.

None of the cmovs became optimal yet sadly. That'll be a follow up PR.
~~Needs #3099 merged first.~~

### CODE REVIEW

**Comment by alyssarosenzweig** [comment]
> A bunch of flag testing is still doing a ubfx plus compare against zero
when it could end up being a tst instead,

These should actually be csel/cset with the host NZCV flags but this requires RA changes (planned but waiting for the rewrite which is the main reason I've ignored cmov's garbage codegen thus far)

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.h:873]: I don't know if this is the right approach. The existing code implements a standard pattern *local value numbering* ... would it be practical to simply augment with a bool `NZCVDirty` instead of duplicating the node?

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1279]: Please redefine this in terms of the explicit size version in this commit to avoid the duplication?

File: unittests/InstructionCountCI/Secondary.json
Comment (alyssarosenzweig) [inline unittests/InstructionCountCI/Secondary.json:276]: Is this bfxil possible to deal with somehow?

Comment (Sonicadvance1) [reply unittests/InstructionCountCI/Secondary.json:276]: Sadly this bfxil can't be removed since 16-bit operations insert in to the GPR instead of zero extending.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1279]: The duplication is a necessary evil while we take steps to remove the usage of implicit sized SelectCC.
Once all implicit uses are removed then the function will get deleted.
Trying to take the conversion slower than the previous PR so it doesn't turn in to a broken mess in one commit.
Just a duplication mess while stuff is changing.

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h:873]: I can switch over to a dirty flag instead

File: unittests/InstructionCountCI/Secondary.json
Comment (alyssarosenzweig) [reply unittests/InstructionCountCI/Secondary.json:276]: Right, ok... that's indeed unfortunate but nothing to be done I suppose

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h:1279]: fair enough

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.h
Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h:873]: Thanks, I suspect that will make more sense, though I'm not super confident tbh

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h]: Rewrote to use a dirty flag to track the writes instead.

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.h]: Thanks, the new commit makes a lot more sense to me! ... CI doesn't seem to agree yet but once it does this will get my approval :+1:

**Review by alyssarosenzweig — APPROVED** [review]

**Pull request merged**
